osu.Game/Rulesets/RulesetStore.cs

```diff
@@ -48,7 +53,17 @@ public RulesetStore(IDatabaseContextFactory factory)
         /// </summary>
         public IEnumerable<RulesetInfo> AvailableRulesets { get; private set; }
 
-        private Assembly resolveRulesetAssembly(object sender, ResolveEventArgs args) => loadedAssemblies.Keys.FirstOrDefault(a => a.FullName == args.Name);
+        private Assembly resolveRulesetDependencyAssembly(object sender, ResolveEventArgs args)
+        {
+            var asm = new AssemblyName(args.Name);
+
+            // this assumes the only explicit dependency of the ruleset is the game core assembly.
+            // the ruleset dependency on the game core assembly requires manual resolving, transient dependencies should be resolved automatically
```

Comment:
So this is because the rulesets are in a subfolder, right? Might as well mention that.